pokemon-app/src/components/EnemyPokemon.js

```diff
@@ -2,101 +2,46 @@ import React, { useState, useEffect } from "react";
-      // Find that type in typeJson, it returns it as a single indexed array,
-      // and so we care only to get the object in the pokemon's type property
-      // which is in that single indexed (this is so we could destructure).
-      typeJson.filter((typeJsonValue) => typeJsonValue[typeValue.type.name])[0][
-        typeValue.type.name
-      ];
-
-    // Return it as an object of these four properties with the values as an array
-    // (this is so we could check if the enemy pokemon's move(s) are super
-    // effective, ineffective, or neutral against the friendly pokemon).
-    return {
-      strongAgainst: strongAgainst.split(", "),
-      weakAgainst: weakAgainst.split(", "),
-      resistantTo: resistantTo.split(", "),
-      vulnerableTo: vulnerableTo.split(", "),
-    };
-  });
-
-  // This function returns an array of integers. It gets the enemy pokemon's
-  // moves and creates a counter based on if it is effective against the
-  // friendly pokemon or not.
-  const moveEffectivenessCounter = enemyPokeMoveTypes.map((moveType) => {
-    let effectiveCounter = 0;
-
-    friendlyTypeMatchups.forEach((typeMatchups) => {
-      // If the enemy's move type is not effective against the friendly pokemon,
-      // then decrement.
-      if (typeMatchups.strongAgainst.includes(capitalize(moveType.name))) {
-        effectiveCounter -= 2;
-        // If the enemy's move type is effective against the friendly pokemon,
-        // then increment.
-      } else if (typeMatchups.weakAgainst.includes(capitalize(moveType.name))) {
-        effectiveCounter += 2;
-        // If the enemy's move type is a standard move type, then add 1.
-      } else {
-        effectiveCounter += 1;
-      }
-    });
-
-    return effectiveCounter;
-  });
-
-  // This returns the array of strings based on the effectiveness counter.
-  return moveEffectivenessCounter.map((effectiveCounter) => {
-    if (effectiveCounter >= 2) {
-      return "Super Effective";
-    } else if (effectiveCounter <= 0) {
-      return "Ineffective";
-    }
-
-    return "";
-  });
-};
+import {
+  moveEffectiveness,
+  overallEffectiveness,
+} from "../utils/effectiveness";
 
 const EnemyPokemon = ({
   enemyPokemon,
   hasEnemy,
   enemyPokeSoundUrl,
   enemyPokeMoveTypes,
   friendlyPokeType,
+  enemyPokeType,
 }) => {
-  const [enemyStatsOnTop, setEnemyStatsOnTop] = useState(true);
+  const [enemyStatsOnTop, setEnemyStatsOnTop] = useState(false);
   const [effectivenessArray, setEffectivenessArray] = useState([]);
+  const [overallEffectivenessString, setOverallEffectivenessString] =
+    useState("");
 
   // This checks if the moves property exist in pokemon object.
   const checkDoesMovesKeyExistInObject = () =>
     Object.prototype.hasOwnProperty.call(enemyPokemon, "moves");
 
   useEffect(() => {
-    if (enemyPokeMoveTypes && friendlyPokeType) {
+    if (
+      enemyPokeMoveTypes.length &&
+      friendlyPokeType.length &&
+      enemyPokeType.length
+    ) {
```

Comment:
Just to clarify, does this check if there are things in the array?

Reply:
Yes. The `.length` would come out as 0 if it were empty (same for strings), and using that in an if-statement would evaluate to false in code language i.e. 0 is false while 1 is true. So, there needs to be something lol